### PR TITLE
DDO-1189 Move build job from Jenkins to Github Actions

### PR DESCRIPTION
**Pull request by mikeflinn**
This pr aims to port the current jenkins build job used for the terra-argocd deployment stack to github actions. The new Job follows the same pattern for build jobs used by other new Terra services such as Workspace manager. It still follows the tagging conventions used in the jenkins job. Pushes to develop are tagged with `dev` and the 12 character sha. 

[example image build off pr branch](https://console.cloud.google.com/gcr/images/broad-dsp-gcr-public/GLOBAL/duos-ui@sha256:77757b6e5839ecc9204771a76a321491abecf2c1cadec1418b6e3d525135582c/details?tab=info&project=broad-dsp-gcr-public)

Additional features include:
- success/failure notifications to #duos-notifcations for builds on `develop` branch
- Automatically updates application version in terra-helmfile for pushes to `develop` 

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] PR is labeled with a Jira ticket number and includes a link to the ticket
- [ ] PR is labeled with a security risk modifier [no, low, medium, high] 
- [x] PR describes scope of changes

In all cases:

- [ ] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [ ] Get PO sign-off for all non-trivial UI or workflow changes
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by rushtong — APPROVED** [review]
This is great! When you merge this, we can delete the jenkins script and job too. Would love to have this for consent/ontology, but with this as a model, I can probably take that on.

**Comment by mikeflinn** [comment]
> This is great! When you merge this, we can delete the jenkins script and job too. Would love to have this for consent/ontology, but with this as a model, I can probably take that on.

Thanks for the thumb, I'm still doing some tweaking to make sure this is actually configured properly. I'll link you some examples when I get it running.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+          vault:1.1.0 \
+          vault write -field token \
+            auth/approle/login role_id=${{ secrets.VAULT_APPROLE_ROLE_ID }} \
+            secret_id=${{ secrets.VAULT_APPROLE_SECRET_ID }})
+        echo ::set-output name=vault-token::$VAULT_TOKEN
+        echo ::add-mask::$VAULT_TOKEN
+    - name: Get Vault Secrets
+      id: vault-secret-get
+      run: |
+        GCR_KEY=$(docker run --rm --cap-add IPC_LOCK \
```
Comment (choover-broad) [inline .github/workflows/build.yaml:46]: This is definitely not a blocker for merging, but we should be trying to move away from having GitHub actions runners pull secrets from Vault directly. (See discussion [here](https://broadinstitute.slack.com/archives/CADM7MZ35/p1615848204267100?thread_ts=1615844943.267000&cid=CADM7MZ35) and [here](https://broadinstitute.slack.com/archives/CQ6SL4N5T/p1615822353013300).)

Hoping we can soon move to using Terraform to pull secrets from Vault and [add them to repos](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/actions_secret) that need them instead. 🤞

**Review by choover-broad — APPROVED** [review]
Nice work! It's a bit awkward trying to do this work now (we want to stop having GHA runners access Vault, AppSec wants us to migrate off broad-dsp-gcr-public to dsp-artifact-registry), but I agree that matching a pattern that already exists, even if it has issues, is better than trying to invent something new.

**Review by rushtong — APPROVED** [review]
Thank you!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (mikeflinn) [reply .github/workflows/build.yaml:46]: Thanks, fixed!

**Comment by mflinn-broad** [comment]
@rushtong Do you need me to fix the failing npm audit check before merging this?

**Pull request merged**